pkg/experimentcalculator/experimentcalc/experiment_calculator.go

```diff
@@ -0,0 +1,660 @@
+	for _, goalID := range experiment.GoalIds {
+		goalResult := &eventcounter.GoalResult{
+			GoalId: goalID,
+		}
+		for _, v := range experiment.Variations {
+			goalResult.VariationResults = append(goalResult.VariationResults, &eventcounter.VariationResult{
+				VariationId: v.Id,
+			})
+		}
+		for _, timestamp := range endAts {
+			evalVc, evalErr := e.getEvaluationCount(ctx, &eventcounter.GetExperimentEvaluationCountRequest{
+				EnvironmentNamespace: envNamespace,
+				StartAt:              experiment.StartAt,
+				EndAt:                timestamp,
+				FeatureId:            experiment.FeatureId,
+				FeatureVersion:       experiment.FeatureVersion,
+				VariationIds:         variationIDs,
+			})
+			if evalErr != nil {
+				e.logger.Error("ExperimentCalculator failed to get evaluation count",
+					log.FieldsFromImcomingContext(ctx).AddFields(
+						zap.String("namespace", envNamespace),
+						zap.String("experiment_id", experiment.Id),
+						zap.Error(evalErr),
+					)...,
+				)
+				return nil, errFailedToGetEvalVariationCount
+			}
+			goalVc, goalErr := e.getGoalCount(ctx, &eventcounter.GetExperimentGoalCountRequest{
+				EnvironmentNamespace: envNamespace,
+				StartAt:              experiment.StartAt,
+				EndAt:                timestamp,
+				GoalId:               goalID,
+				FeatureId:            experiment.FeatureId,
+				FeatureVersion:       experiment.FeatureVersion,
+				VariationIds:         variationIDs,
+			})
+			if goalErr != nil {
+				e.logger.Error("ExperimentCalculator failed to get goal count",
+					log.FieldsFromImcomingContext(ctx).AddFields(
+						zap.String("namespace", envNamespace),
+						zap.String("experiment_id", experiment.Id),
+						zap.Error(goalErr),
+					)...,
+				)
+				return nil, errFailedToGetGoalEventCount
+			}
+			gr := e.calcGoalResult(ctx, evalVc, goalVc, experiment.BaseVariationId)
+			gr.GoalId = experiment.GoalId
+			e.appendVariationResult(ctx, timestamp, goalResult, gr.VariationResults)
+		}
+		experimentResult.GoalResults = append(experimentResult.GoalResults, goalResult)
+	}
+
+	return experimentResult, nil
+}
+
+func (e ExperimentCalculator) listEnvironments(
+	ctx context.Context,
+) ([]*environment.Environment, error) {
+	listEnvironmentsRequest := environment.ListEnvironmentsRequest{
+		PageSize: 0,
+		Cursor:   "",
+	}
+	resp, err := e.environmentClient.ListEnvironments(ctx, &listEnvironmentsRequest)
+	if err != nil {
+		return nil, err
+	}
+	return resp.Environments, err
+}
+
+func (e ExperimentCalculator) listExperiments(
+	ctx context.Context,
+	namespace string,
+) ([]*experiment.Experiment, error) {
+	req := &experiment.ListExperimentsRequest{
+		From:                 time.Now().In(e.location).Add(-2 * 24 * time.Hour).Unix(),
+		PageSize:             0,
+		Cursor:               "",
+		EnvironmentNamespace: namespace,
+		Statuses: []experiment.Experiment_Status{
+			experiment.Experiment_WAITING,
```

Comment:
@Ubisoft-potato, I just noticed that because we will start the experiment via batch service, we no longer need to get the experiment in WAITING status.

Also, we can remove this condition.
https://github.com/bucketeer-io/bucketeer/pull/395/files#diff-b0f1a6e9d024e46f089e3bc6573a1c8f6dfaf821d6b3898e3c06794bbc4d0dd8R128-R131

Reply:
Done!